src/Engines/AlgoliaEngine.php

```diff
@@ -134,4 +139,37 @@ public function map($results, $model)
             return $models[$hit[$model->getKeyName()]];
         });
     }
+
+    public function setSettings($model, $settings)
+    {
+        $index = $this->algolia->initIndex($model->searchableAs());
+
+        if (isset($settings['synonyms'])) {
+            $index->batchSynonyms($settings['synonyms'], true, true);
```

Comment:
I will just chunk it now. I don't want to play with Scout APIs too much as it might be blocking point.